stdlib/public/Concurrency/Executor.swift

```diff
@@ -577,9 +584,8 @@ extension Task where Success == Never, Failure == Never {
   /// executor is a fatal error.
   @available(SwiftStdlib 6.2, *)
   public static var defaultExecutor: any TaskExecutor {
-    if _defaultExecutor == nil {
-      _defaultExecutor = DefaultExecutorFactory.defaultExecutor
-    }
+    // It would be good if there was a Swift way to do this
```

Comment:
Wouldn't something like this get thread-safe lazy initialization? I believe we do that for `static let`s.

```swift
public static let defaultExector: any TaskExecutor = DefaultExecutorFactory.defaultExecutor
```

Reply:
It's a bit more complicated than it looks :-)

Yes, initialising a static variable does happen in a thread-safe manner, but we don't actually want to initialise these to the default executor (since that means constructing the default executor always, even when we don't actually want it, and the act of constructing it _might_ have side effects).

Reply:
I think we could leverage static variable initialisation, but I think to do so we'd need to add an extra variable and make the static initialisation be a closure call. I'm not sure that's really an improvement overall. What _would_ be an improvement is the ability to do a `once` explicitly and directly in Swift, since that's what we really want here.